### PR TITLE
Stream-K Sample Validation

### PR DESCRIPTION
**Pull request by dlangbe**
Fixed validation issues with streamk_hgemm sample.

### CODE REVIEW

**Review by neoblizz — APPROVED** [review]

File: samples/streamk_hgemm.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ using MfmaFragA   = fragment<matrix_a, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, InputT,
 using MfmaFragB   = fragment<matrix_b, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, InputT, DataLayoutB>;
 using MfmaFragC   = fragment<accumulator, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, OutputT, DataLayoutC>;
 using MfmaFragD   = MfmaFragC;
-using MfmaFragAcc = fragment<accumulator, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, ComputeT>;
+using MfmaFragAcc = fragment<accumulator, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, ComputeT, DataLayoutC>;
```
Comment (cgmillette) [inline samples/streamk_hgemm.cpp:241]: Accumulator is typically in ComputeT. Why DataLayoutC?

File: samples/streamk_hgemm.cpp [hunk 2]
```diff
@@ -404,6 +404,32 @@ ROCWMMA_DEVICE static inline void
     }
 }
 
+// Global Acc reads used in partial fixup step
+ROCWMMA_DEVICE static inline void
+    globalReadAcc(MfmaFragAcc (&fragsAcc)[BLOCKS_X][BLOCKS_Y], ComputeT const* gAddrAcc, uint32_t ldAcc)
+{
+    // using MfmaFragTemp = fragment<accumulator, ROCWMMA_M, ROCWMMA_N, ROCWMMA_K, ComputeT, DataLayoutC>;
```
Comment (cgmillette) [inline samples/streamk_hgemm.cpp]: Can remove comment

File: samples/streamk_hgemm.cpp [hunk 3]
```diff
@@ -404,6 +404,32 @@ ROCWMMA_DEVICE static inline void
     }
 }
 
+// Global Acc reads used in partial fixup step
+ROCWMMA_DEVICE static inline void
+    globalReadAcc(MfmaFragAcc (&fragsAcc)[BLOCKS_X][BLOCKS_Y], ComputeT const* gAddrAcc, uint32_t ldAcc)
```
Comment (cgmillette) [inline samples/streamk_hgemm.cpp:409]: Data address in ComputeT. See above definition of MfmaFragAcc datatype.

**Review by cgmillette — APPROVED** [review]
Looks good to me!

**Pull request merged**